searx/plugins/limiter.py

```diff
@@ -36,8 +36,9 @@
 )
 
 
-def is_accepted_request(inc_get_counter) -> bool:
+def is_accepted_request() -> bool:
     # pylint: disable=too-many-return-statements
+    redis_client = redisdb.client()
```

Comment:
The Redis client implements:
* a retry helper: https://redis.readthedocs.io/en/v4.3.1/retry.html
* a connection pool: https://pypi.org/project/redis/ (see "Connection Pools")

IMO, we can create one client and share it between the different threads.

So, for example, `incr_sliding_window(client, name: str, duration: int)` becomes `incr_sliding_window(name: str, duration: int)`.

Reply:
I know, but I think it is better not to bind a library to *one* DB connection.  IMO: The management of the DB connections is out of the scope of a library that implements of functions. .. may be one day we have multiple connections or a plugin want to use its own connection.

Reply:
* `redisdb.client()` connects to `localhost:6379`.
* the instantiation of a Redis client for each request has a performance impact according to [this micro benchmark](https://gist.github.com/dalf/d43a06ad4e1c5c3913ff57b3278da22a#file-reuse_redis_client-py)

Reply:
> * `redisdb.client()` connects to `localhost:6379`.

It is connected to whatever redis.url from settings point to: `redis.Redis.from_url(get_setting('redis.url')) `

> * the instantiation of a Redis client for each request has a performance impact

Not sure I understand, but why do you think that we *"the instantiation of a Redis client for each request"* .. if you look at the code sequence shown above, there is a global variable `_client` that is set only once (if it is None).

Or do I missed something?

Reply:
I'm sorry I've read `redis.Client()` instead of `redisdb.client()`